src/coreclr/CMakeLists.txt

```diff
@@ -34,8 +36,8 @@ OPTION(CLR_CMAKE_ENABLE_CODE_COVERAGE "Enable code coverage" OFF)
 OPTION(CLR_CMAKE_WARNINGS_ARE_ERRORS "Warnings are errors" ON)
 
 # Ensure other tools are present
-if (WIN32)
-    if(CLR_CMAKE_HOST_ARCH STREQUAL arm)
+if (CLR_CMAKE_TARGET_WIN32)
```

Comment:
Should be controlled by `CLR_CMAKE_HOST_*` variable not target.  These set compiler settings.